### ISSUE TITLE
Network error: Unexpected end of JSON input

### ISSUE THREAD

**Issue by imgntn** [Shopify/quilt#1419]
## Overview

When I clone repo and follow all setup steps for the repository https://shopify.dev/tutorials/build-a-shopify-app-with-node-and-react/, I get errors on my index page.

It displays this error on the page
`"Network error: Unexpected end of JSON input" `

console shows
```
index.js:1 Warning: Did not expect server HTML to contain a <div> in <div>.

bundle.esm.js:69 POST https://jbpshopify2.ngrok.io/graphql 406 (Not Acceptable)
```

I haven't changed any code, only added a .env file that looks like this

```
SHOPIFY_API_KEY=x
SHOPIFY_API_SECRET_KEY=x
HOST=https://jbpshopify2.ngrok.io
```

Any tips on what to do to get it working?  Thank you.

...

## Consuming repo

> What repo were you working in when this issue occurred?

https://github.com/Shopify/shopify-demo-app-node-react

There's unfortunately no issues on that repository so I thought I'd ask here.

---

## Checklist

- [x ] Please delete the labels section before submitting your issue
- [ x] I have described this issue in a way that is actionable (if possible)

**Comment by imgntn** [comment]
Sorted it out - there were somehow malformed "products" that had been stored in the localStorage of the embedded app, once those were cleared out it started to work again!  Thanks for letting me think aloud :)

**Comment by Eurotimmy** [comment]
Hey there @imgntn 
Thank you for thinking aloud as I have exactly the same issue in the same shopify repo.
May I ask how you resolved this error please?

**Comment by imgntn** [comment]
I went to the Developer Tools in Chrome -> Application Tab -> Storage section -> Local Storage.  Then I deleted the entries there and restarted my app and it (thankfully) worked.

**Comment by Eurotimmy** [comment]
@imgntn  Thank you for replying.
Unfortunately this didn't work for me... it may be because I am running the code on an AWS instance.
I'll keep messing about with it and hope somebody else might chime in here if they know other things to try.

**Comment by jbkhellas** [comment]
ok, i will also try my luck by thinking out aloud...
facing the same problem, the local storage seems ok:
ids: ["gid://shopify/Product/5307389935776","gid://shopify/Product/5307390591136"]
yet i still get the same error you guys get...

**Comment by Eurotimmy** [comment]
Hello @jbkhellas, @imgntn and others who may stumble upon this in future.
Myself and another user I work with installed this repo on the same day back in June, we both experienced the error outlined in the OP... this > "Network error: Unexpected end of JSON input"
The same error was shown in each of our AWS and Heroku accounts when we tried to get the demo to work.

Yesterday (26 July 2020) we cloned the current version of the '[shopify-demo-app-node-react](https://github.com/Shopify/shopify-demo-app-node-react)' repo and with the minimal required changes it works correctly.
Both AWS and Heroku environments are working for us.
